azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -331,78 +338,220 @@ def __init__(self, xml_text):
+          <DependsOnExtension extension="firstRunCommand" handler="Microsoft.CPlat.Core.RunCommandHandlerWindows" />
+          <DependsOnExtension handler="Microsoft.Compute.CustomScriptExtension" />
+          <DependsOnExtension handler="Microsoft.Compute.VMAccessAgent" />
+        </DependsOn>
+        <DependsOn dependencyLevel="4" name="thirdRunCommand">
+          <DependsOnExtension extension="firstRunCommand" handler="Microsoft.CPlat.Core.RunCommandHandlerWindows" />
+          <DependsOnExtension extension="secondRunCommand" handler="Microsoft.CPlat.Core.RunCommandHandlerWindows" />
+          <DependsOnExtension handler="Microsoft.Compute.CustomScriptExtension" />
+          <DependsOnExtension handler="Microsoft.Compute.VMAccessAgent" />
+        </DependsOn>
+        <ExtensionRuntimeSettings seqNo="2" name="firstRunCommand" state="enabled">
+            {
+              "runtimeSettings": [
+                {
+                  "handlerSettings": {
+                    "publicSettings": {"source":{"script":"Write-Host First: Hello World 1234!"}}
+                  }
+                }
+              ]
+            }
+        </ExtensionRuntimeSettings>
+        <ExtensionRuntimeSettings seqNo="2" name="secondRunCommand" state="enabled">
+            {
+              "runtimeSettings": [
+                {
+                  "handlerSettings": {
+                    "publicSettings": {"source":{"script":"Write-Host First: Hello World 1234!"}}
+                  }
+                }
+              ]
+            }
+        </ExtensionRuntimeSettings>
+        <ExtensionRuntimeSettings seqNo="1" name="thirdRunCommand" state="enabled">
+            {
+              "runtimeSettings": [
+                {
+                  "handlerSettings": {
+                    "publicSettings": {"source":{"script":"Write-Host Third: Hello World 3!"}}
+                  }
+                }
+              ]
+            }
+        </ExtensionRuntimeSettings>
+      </Plugin>
+        """
+        # Parse and cache the Dependencies for each extension first
+        dependency_levels = defaultdict(int)
+        for depends_on_node in findall(plugin_settings_node, "DependsOn"):
+            extension_name = getattrib(depends_on_node, "name")
+            if not ExtensionsConfig.__is_not_empty(extension_name):
+                raise ExtensionConfigError("No Name not specified for DependsOn object in ExtensionRuntimeSettings for MultiConfig!")
+
+            dependency_level = ExtensionsConfig.__get_dependency_level_from_node(depends_on_node, extension_name)
+            dependency_levels[extension_name] = dependency_level
+
+        for extension_runtime_setting_node in extension_runtime_settings_nodes:
+            # Name and State will only be set for ExtensionRuntimeSettings for Multi-Config
+            extension_name = getattrib(extension_runtime_setting_node, "name")
+            if not ExtensionsConfig.__is_not_empty(extension_name):
```

Comment:
similar question to above: getattrib always return None if no match? why the additional check against "" and []?

Reply:
Answered above